plugins/catalog-react/report-alpha.api.md

```diff
@@ -332,17 +338,48 @@ export const EntityContextMenuItemBlueprint: ExtensionBlueprint<{
   kind: 'entity-context-menu-item';
   name: undefined;
   params: EntityContextMenuItemParams;
-  output: ConfigurableExtensionDataRef<JSX_2.Element, 'core.reactElement', {}>;
+  output:
+    | ConfigurableExtensionDataRef<JSX_2.Element, 'core.reactElement', {}>
+    | ConfigurableExtensionDataRef<
+        (entity: Entity) => boolean,
+        'catalog.entity-filter-function',
+        {
+          optional: true;
+        }
+      >
+    | ConfigurableExtensionDataRef<
+        string,
+        'catalog.entity-filter-expression',
+        {
+          optional: true;
+        }
+      >;
   inputs: {};
-  config: {};
-  configInput: {};
-  dataRefs: never;
+  config: {
+    filter: EntityPredicate | undefined;
+  };
+  configInput: {
+    filter?: EntityPredicate | undefined;
+  };
+  dataRefs: {
+    filterFunction: ConfigurableExtensionDataRef<
+      (entity: Entity) => boolean,
+      'catalog.entity-filter-function',
+      {}
+    >;
+    filterExpression: ConfigurableExtensionDataRef<
+      string,
+      'catalog.entity-filter-expression',
+      {}
+    >;
+  };
 }>;
 
 // @alpha (undocumented)
 export type EntityContextMenuItemParams = {
   useProps: UseProps;
   icon: JSX_2.Element;
+  filter?: string | EntityPredicate | ((entity: Entity) => boolean);
```

Comment:
Let's skip `string` filters, as in `entityFilterExpressionDataRef`. They're deprecated and will be removed soon anyway. That was we don't need to move `buildFilterFn` either.